src/main/java/at/bitfire/ical4android/validation/ICalPreprocessor.kt

```diff
@@ -59,21 +60,33 @@ object ICalPreprocessor {
      */
     fun preprocessCalendar(calendar: Calendar) {
         for (component in calendar.components)
-            for (property in component.properties)
-                applyRules(property)
+            applyRules(component)
     }
 
     @Suppress("UNCHECKED_CAST")
-    private fun applyRules(property: Property) {
-        propertyRules
-            .filter { rule -> rule.supportedType.isAssignableFrom(property::class.java) }
+    private fun applyRules(component: Component) {
+        // Apply rules to component
+        preprocessorRules
+            .filter { rule -> rule.supportedType.isAssignableFrom(component::class.java) }
             .forEach {
-                val beforeStr = property.toString()
-                (it as Rfc5545PropertyRule<Property>).applyTo(property)
-                val afterStr = property.toString()
+                val beforeStr = component.toString()
+                (it as Rfc5545Rule<Component>).applyTo(component)
+                val afterStr = component.toString()
                 if (beforeStr != afterStr)
                     Ical4Android.log.log(Level.FINER, "$beforeStr -> $afterStr")
             }
+
+        // Apply to properties
+        for (property in component.properties)
+            preprocessorRules
+                .filter { rule -> rule.supportedType.isAssignableFrom(property::class.java) }
+                .forEach {
+                    val beforeStr = property.toString()
+                    (it as Rfc5545Rule<Property>).applyTo(property)
+                    val afterStr = property.toString()
+                    if (beforeStr != afterStr)
+                        Ical4Android.log.log(Level.FINER, "$beforeStr -> $afterStr")
+                }
```

Comment:
We can also revert this for now because we don't need Component rules.

If we need them one day, better use two arrays instead of using `.filter`. For instance I have overlooked this code yesterday because I didn't expect to happen magic filter actions on the array.